src/openzaak/components/zaken/api/validators.py

```diff
@@ -525,3 +527,71 @@ def __call__(self, attrs, serializer):
 
         if zaak.archiefstatus != Archiefstatus.nog_te_archiveren:
             raise serializers.ValidationError(self.message, code=self.code)
+
+
+def match_eigenschap_specificatie(spec, value: str) -> bool:
+    """
+    validate value against eigenschap.specificatie
+    moved to the separate function to reuse for admin validation
+    """
+    # enum
+    if spec.waardenverzameling:
+        return value in spec.waardenverzameling
+
+    if spec.formaat == FormaatChoices.tekst:
+        return len(value) <= int(spec.lengte)
+
+    if spec.formaat == FormaatChoices.getal:
+        whole_length = spec.lengte.split(",")[0]
+        regex = rf"\d{{0,{whole_length}}}"
+        if "," in spec.lengte:
+            fractional_length = spec.lengte.split(",")[-1]
+            regex += rf",?\d{{0,{fractional_length}}}"
+
+        match = re.fullmatch(regex, value)
+        return bool(match)
+
+    if spec.formaat == FormaatChoices.datum:
+        # according ZGW standard datum should be in 'jjjjmmdd' format
+        try:
+            datetime.strptime(value, "%Y%m%d")
+        except ValueError:
+            return False
+        else:
+            return True
+
+    if spec.formaat == FormaatChoices.datum_tijd:
+        # according ZGW standard datum/tijd should be in 'jjjjmmdduummss' format
+        try:
+            datetime.strptime(value, "%Y%m%d%H%M%S")
+        except ValueError:
+            return False
+        else:
+            return True
+
+    return True
+
+
+class ZaakEigenschapValueValidator:
+    code = "waarde-incorrect-format"
+    message = _("The 'waarde' value doesn't match the related eigenschap.specificatie")
```

Comment:
Maybe this should be more specific and mention what the expected format is i.e. `does not match eigenschap.specificatie datum [8, #1]` or `str(spec)`

Reply:
Good idea. I've changed the message